tests/pipelines/test_pipelines_image_segmentation.py

```diff
@@ -259,25 +266,43 @@ def test_small_model_pt(self):
         # The `panoptic` returns only LABEL_215, and this returns 3 labels.
         #
         output = image_segmenter("http://images.cocodataset.org/val2017/000000039769.jpg", subtask="semantic")
+
+        output_masks = [o["mask"] for o in output]
+        expected_masks = [
+            "https://huggingface.co/datasets/hf-internal-testing/mask-for-image-segmentation-tests/resolve/main/mask_0.png",
```

Comment:
Do you mind use `load_dataset` to get the actual masks, and put in comments the urls:
https://datasets-server.huggingface.co/assets/hf-internal-testing/mask-for-image-segmentation-tests/--/hf-internal-testing--mask-for-image-segmentation-tests/train/4/image/image.jpg

So that readers of the test can get a look at the mask (not download them)

Reply:
Also could we have the masks for the slow tests ? That would help demonstrate the readability improvement

Reply:
For `load_dataset`, I am not very sure what's the best way to use it. Should I use
```python
target_mask_0 = dataset[0]
target_mask_1 = dataset[1]
target_mask_2 = dataset[2]
```
?? I don't feel super comfortable to use the order (indices) in the datasets (i.e. assuming the order in dataset and in the file name correspond perfectly.

I think I can use similar links (the format you provided above) for both visualization as well as for `requests.get` in the testing method. **Would this be OK for you**? 

Regarding masks for other tests (slow or not), I can upload them progressively, and replace the tests to use the new way of checking threshold.

Reply:
Well , the link format doesn't use the original file name 😢

Reply:
> ?? I don't feel super comfortable to use the order (indices) in the datasets (i.e. assuming the order in dataset and in the file name correspond perfectly.

You could get the name indexing doing something like:
```python
data = {item["filename"]: item["image"] for item in dataset}
```
for instance ?
You could add back the filename too.

The advantage of using `datasets` is that we don't have to download the images on every single test.

It's not a super important thing, just trying to reuse our tools and limit slightly network. Either solution is fine.

Reply:
Mmm I don't see the point of downloading and caching the whole dataset if we just need three images in it? (I know it only has 9 images, but still).

Reply:
Let me see if I can make the image visualization easier (i.e. not downloading the image when clicking the links). Regarding if we want to use `load_dataset`, let see when I need to upload more images to the Hub (for more segmentation tests) and use them in those tests.